### PR TITLE
Fixes error handling in has_session_key function

### PR DESCRIPTION
**Pull request by jonathanudd**

### CODE REVIEW

**Comment by Maharacha** [comment]
just thinking.. how about throwing an exception with the message instead?
so return:
True if key exist
False if key does not exist
Throw exception on error

**Comment by jakobilobi** [comment]
> @sedrik linked this to me https://github.com/rustedpy/result. Maybe we should try this?

Looks cool and useful :ok_man:

File: src/polkadot_rpc_wrapper.py [hunk 1]
```diff
@@ -109,33 +111,39 @@ def find_validator_address(self):
...
-                return {"validator": validator[0], "session_key": session_key}
-        return False
+            has_session_key, error = self.has_session_key(session_key)
+            if error:
+                return False, error
+            if has_session_key:
+                return {"validator": validator[0], "session_key": session_key}, ""
+        return False, ""
 
     def is_validating_next_era(self, address):
```
Comment (jakobilobi) [inline src/polkadot_rpc_wrapper.py:121]: Add return types for all functions?

You did for some, but not for all :man_shrugging:

File: src/polkadot_rpc_wrapper.py [hunk 2]
```diff
@@ -73,17 +73,19 @@ def get_system_peers(self) -> Tuple[list, bool]:
...
+        :return: Tuple[bool, str]
         """
         data = '{"id": 1, "jsonrpc":"2.0", "method": "author_hasSessionKeys", "params":["' + session_key + '"]}'
         response = requests.post(url=self.__server_address, headers=self.__headers, data=data)
         response_json = json.loads(response.text)
+        if 'error' in response_json.keys():
+            return False, response_json['error']['message']
         result = response_json['result']
-        return result
+        return result, ""
```
Comment (jakobilobi) [inline src/polkadot_rpc_wrapper.py:88]: I believe the Pythonic way to handle "returning" an error isn't to do it like this, but to raise an actual `Error` and then except it in the function above (or pass it on up, depending on what the aim is). I'm not saying we can't do it like this, especially with a cool tool like the package you linked, but there are other ways too.

```suggestion
    def has_session_key(self, session_key) -> bool:
        """
        Checks if this node has the supplied session_key (E.g. 0xb75f94a5eec... )
        :param session_key: string
        :return: bool
        """
        data = '{"id": 1, "jsonrpc":"2.0", "method": "author_hasSessionKeys", "params":["' + session_key + '"]}'
        response = requests.post(url=self.__server_address, headers=self.__headers, data=data)
        response_json = json.loads(response.text)
        if 'error' in response_json.keys():
            raise SomeCustomError(response_json['error']['message'])
        return response_json['result']
```

This comment applies to all functions where you added a tuple for errors.

**Comment by jakobilobi** [comment]
So how are things going here? Let's get this into `develop` perhaps? :)

**Pull request closed without merging**